src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -150,5 +155,96 @@ public void ConnectHeaders_NewAndExistingHeaders_ShouldCreateNewHeaders()
+
+            var chainTip = testContext.ExtendAChain(10);
+            cht.Initialize(chainTip, true);
+            chainTip.BlockDataAvailability = BlockDataAvailabilityState.BlockAvailable;
+            ChainedHeader newChainTip = testContext.ExtendAChain(10, chainTip);
+
+            var listOfExistingHeaders = testContext.ChainedHeaderToList(chainTip, 10);
+            var listOfNewHeaders = testContext.ChainedHeaderToList(newChainTip, 10);
+
+            testContext.ChainStateMock.Setup(s => s.ConsensusTip).Returns(chainTip);
+            chainTip.BlockValidationState = ValidationState.FullyValidated;
+
+            cht.ConnectNewHeaders(1, listOfExistingHeaders);
+
+            var peerIdsByTipHashBefore = new Dictionary<uint256, HashSet<int>>(cht.GetPeerIdsByTipHash());
+            var chainedHeaderTreeHashesBefore = cht.GetChainedHeadersByHash();
+
+            cht.ConnectNewHeaders(1, listOfNewHeaders);
+
+            var peerIdsByTipHashAfter = cht.GetPeerIdsByTipHash();
+            var chainedHeaderTreeHashesAfter = cht.GetChainedHeadersByHash();
+
+            // Same number of entries as we are reassigning hash
+            Assert.True(chainedHeaderTreeHashesBefore.Count == chainedHeaderTreeHashesAfter.Count );
+
+            // Peer's tip hash has changed
+            Assert.True(peerIdsByTipHashBefore.FirstOrDefault(x => x.Value.Contains(1)).Key !=
+                        peerIdsByTipHashAfter.FirstOrDefault(x => x.Value.Contains(1)).Key);
+        }
+
+        // Make sure checkpoints are off - supply some headers,
+        // CHT should return ToDownload array of the same size as the amount of headers.
+        [Fact]
+        public void ConnectHeaders_SupplyHeaders_ToDownloadArraySizeSameAsNumberOfHeaders()
+        {
+            // Setup
+            var ctx = new TestContext();
+            ChainedHeaderTree cht = ctx.CreateChainedHeaderTree();
+            var chainTip = ctx.ExtendAChain(10);
+            cht.Initialize(chainTip, true);
+            chainTip.BlockDataAvailability = BlockDataAvailabilityState.BlockAvailable;
+            ctx.ChainStateMock.Setup(s => s.ConsensusTip).Returns(chainTip);
+            ctx.ConsensusSettings.UseCheckpoints = false;
+            ChainedHeader newChainTip = ctx.ExtendAChain(10, chainTip);
+            List<BlockHeader> listOfNewHeaders = ctx.ChainedHeaderToList(newChainTip, 10);
+
+            // Checkpoints are off
+            Assert.False(ctx.ConsensusSettings.UseCheckpoints);
+
+            // Supply some headers
+            var connectedNewHeaders = cht.ConnectNewHeaders(PEER_ONE, listOfNewHeaders);
+
+            var chainedHeaderFrom = connectedNewHeaders.DownloadFrom;
+            var chainedHeaderTo = connectedNewHeaders.DownloadTo;
+            int headersToDownloadCount = chainedHeaderTo.Height - chainedHeaderFrom.Height + 1; // Inclusive
+
+            // ToDownload array of the same size as the amount of headers
```

Comment:
Little bit outdated to description again, that is why I said it might be good thing to rewrite it. No longer we use to download array, but as you correctly recognised we only transfer lower and upper bounds. So the in-line comments should reflect the reality.

Reply:
I'm not sure which pattern or code construct replaces DownloadTo?  Happy to make these changes

Reply:
You have it correctly in the code, just the comment